dp/max_product_subarray.py

```diff
@@ -19,3 +19,38 @@ def max_product(nums):
         lmax = max(max(t1, t2), nums[i])
         lmin = min(min(t1, t2), nums[i])
         gmax = max(gmax, lmax)
+
+
+'''
+Another approach that would print max product and the subarray
+
+Example:  
+subarray_with_max_product([2,3,6,-1,-1,9,5])
+max_product_so_far: 45, [9, 5]
+
+subarray_with_max_product([2,3,6,-1,-1,4,5])
+max_product_so_far: 36, [2, 3, 6]
+
+subarray_with_max_product([-2,-3,6,-1,-9,-5])
+max_product_so_far: 6, [6]
+'''
+
+def subarray_with_max_product(arr):
+    ''' arr is list of positive/negitive numbers '''
+    l = len(arr)
+    product_so_far = max_product_end = 1
+    max_start_i = 0
+    so_far_start_i = so_far_end_i = 0
+
+    for i in range(l):
+        max_product_end *= arr[i]
+        if max_product_end < 0:
+            max_product_end = 1
+            max_start_i = i + 1
+
+        if product_so_far < max_product_end:
+            product_so_far = max_product_end
+            so_far_end_i = i
+            so_far_start_i = max_start_i
+    print "max_product_so_far: %s, %s" % (product_so_far,\
+	arr[so_far_start_i:so_far_end_i + 1])
```

Comment:
Please indent this line according to opening parenthesis (as per pep8 guidelines)